Tests/SmallDataIOTest/SmallDataIOTest.cpp

```diff
@@ -0,0 +1,181 @@
+
+    static const std::vector<std::string> header1_strings{"x", "y", "z"};
+
+    // write a header containing labels for each column
+    test_file.write_header_line(header1_strings); // pre-header string is time
+
+    for (int i = 0; i < col[0].size(); ++i)
+    {
+        std::vector<amrex::Real> some_data{col[0][i], col[1][i], col[2][i]};
+        test_file.write_time_data_line(some_data);
+    }
+}
+
+std::vector<SmallDataIO::column_t>
+test_small_data_io_reader(const std::vector<std::string> &column_names)
+{
+
+    SmallDataIO test_reader("test_000000");
+
+    // Could print out file structure as well
+    // test_reader.print_file_structure();
+
+    std::vector<std::string> header;
+    test_reader.get_header_strings(header, 0);
+
+    // Columns do not have to be in order, you could also have entered
+    // {"z", "x"} for example. But data[0] will always go with the
+    // first entry, data[1] with the second etc.
+
+    // If a column name doesn't exist in the header, amrex::Abort will
+    // be called
+
+    std::vector<SmallDataIO::column_t> data;
+    test_reader.get_columns(data, column_names, 0);
+
+    SmallDataIO::broadcast_data(data);
+
+    return data;
+}
+
+// NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
+std::vector<SmallDataIO::column_t>
+test_small_data_io_reader(const int a_min_col, const int a_max_col)
+{
+
+    SmallDataIO test_reader("test_000000");
+
+    std::vector<SmallDataIO::column_t> data;
+    test_reader.get_columns(data, a_min_col, a_max_col, 0);
+
+    SmallDataIO::broadcast_data(data);
+
+    return data;
+}
+
+std::vector<double> test_small_data_io_reader(const int a_col)
+
+{
+    SmallDataIO test_reader("test_000000");
+
+    std::vector<SmallDataIO::column_t> data;
+    test_reader.get_column(data, a_col, 0);
+
+    return data[0];
+}
+
+void run_small_data_io_test()
+{
+    int amrex_argc    = doctest::cli_args.argc();
+    char **amrex_argv = doctest::cli_args.argv();
+
+    // NOLINTNEXTLINE(bugprone-casting-through-void) // Open MPI triggers this
+    amrex::Initialize(amrex_argc, amrex_argv);
+    {
+
+        // Generate some random numbers to write out
+        static const int Npts{100};
+
+        // This is GPU safe, but must streamSynchronize because of Async copy
+        amrex::ULong cpu_seed = 12345;
+        amrex::ULong gpu_seed = 67890;
+        amrex::InitRandom(cpu_seed, amrex::ParallelDescriptor::NProcs(),
+                          gpu_seed);
+
+        std::vector<SmallDataIO::column_t> write_data(3);
+        for (auto &column : write_data)
+        {
+            column.resize(Npts);
+            column = generate_random_numbers(Npts);
+        }
+
+        amrex::Gpu::streamSynchronize();
+
+        static constexpr int data_precision = 10;
+        static const amrex::Real err_tol = std::pow(10., -1.0 * data_precision);
+
+        // Test the file writer: write out the random co-oridinates
+
+        test_small_data_io_writer(write_data, data_precision);
```

Comment:
Just a minor typo/formatting.
```suggestion
        // Test the file writer: write out the random coordinates
        test_small_data_io_writer(write_data, data_precision);
```